plugins/modules/mso_schema_template_vrf_contract.py

```diff
@@ -0,0 +1,265 @@
+        consumer_contracts = [c.get('contractRef') for c in schema_obj.get('templates')[template_idx]['vrfs'][vrf_idx]['vzAnyConsumerContracts']]
+        contract_ref = mso.contract_ref(**contract)
+        if contract_ref in provider_contracts and contract.get('type') == 'provider':
+            contract_idx = provider_contracts.index(contract_ref)
+            contract_path = '/templates/{0}/vrfs/{1}/vzAnyProviderContracts/{2}'.format(template, vrf, contract_idx)
+            mso.existing = schema_obj.get('templates')[template_idx]['vrfs'][vrf_idx]['vzAnyProviderContracts'][contract_idx]
+        if contract_ref in consumer_contracts and contract.get('type') == 'consumer':
+            contract_idx = consumer_contracts.index(contract_ref)
+            contract_path = '/templates/{0}/vrfs/{1}/vzAnyConsumerContracts/{2}'.format(template, vrf, contract_idx)
+            mso.existing = schema_obj.get('templates')[template_idx]['vrfs'][vrf_idx]['vzAnyConsumerContracts'][contract_idx]
+        if mso.existing.get('contractRef'):
+            mso.existing['contractRef'] = mso.dict_from_ref(mso.existing.get('contractRef'))
+            mso.existing['relationshipType'] = contract.get('type')
+
+    if state == 'query':
+        if not contract:
+            provider_contracts = [dict(contractRef=mso.dict_from_ref(c.get('contractRef')),
+                                  relationshipType='provider') for c in schema_obj.get('templates')[template_idx]['vrfs'][vrf_idx]['vzAnyProviderContracts']]
+            consumer_contracts = [dict(contractRef=mso.dict_from_ref(c.get('contractRef')),
+                                  relationshipType='consumer') for c in schema_obj.get('templates')[template_idx]['vrfs'][vrf_idx]['vzAnyConsumerContracts']]
+            mso.existing = provider_contracts + consumer_contracts
+        elif not mso.existing:
+            mso.fail_json(msg="Contract '{0}' not found".format(contract.get('name')))
+
+        mso.exit_json()
+
+    if contract.get('type') == 'provider':
+        contracts_path = '/templates/{0}/vrfs/{1}/vzAnyProviderContracts'.format(template, vrf)
+    if contract.get('type') == 'consumer':
+        contracts_path = '/templates/{0}/vrfs/{1}/vzAnyConsumerContracts'.format(template, vrf)
+    ops = []
+
+    mso.previous = mso.existing
+    if state == 'absent':
+        if mso.existing:
+            mso.sent = mso.existing = {}
+            ops.append(dict(op='remove', path=contract_path))
+
+    elif state == 'present':
+        payload = dict(
+            contractRef=dict(
+                contractName=contract.get('name'),
+                templateName=contract.get('template'),
+                schemaId=contract.get('schema_id'),
+            ),
+        )
+
+        mso.sanitize(payload, collate=True)
+
+        if mso.existing:
+            ops.append(dict(op='replace', path=contract_path, value=mso.sent))
+        else:
+            ops.append(dict(op='add', path=contracts_path + '/-', value=mso.sent))
```

Comment:
You should probably add the /- directly to contracts_path as it is not used anywhere else so you avoid concatenation here again.